tests/test_transforms/test_inference.py

```diff
@@ -0,0 +1,1664 @@
+
+    @to_be_fixed(raises=Exception)
+    @pytest.mark.parametrize(
+        "transform, dataset_name",
+        [
+            # encoders
+            (MeanSegmentEncoderTransform(), "regular_ts"),
+            (SegmentEncoderTransform(), "regular_ts"),
+            # math
+            (BoxCoxTransform(in_column="target", mode="per-segment", inplace=False), "positive_ts"),
+            (BoxCoxTransform(in_column="target", mode="per-segment", inplace=True), "positive_ts"),
+            (MaxAbsScalerTransform(in_column="target", mode="per-segment", inplace=False), "regular_ts"),
+            (MaxAbsScalerTransform(in_column="target", mode="per-segment", inplace=True), "regular_ts"),
+            (MinMaxScalerTransform(in_column="target", mode="per-segment", inplace=False), "regular_ts"),
+            (MinMaxScalerTransform(in_column="target", mode="per-segment", inplace=True), "regular_ts"),
+            (RobustScalerTransform(in_column="target", mode="per-segment", inplace=False), "regular_ts"),
+            (RobustScalerTransform(in_column="target", mode="per-segment", inplace=True), "regular_ts"),
+            (StandardScalerTransform(in_column="target", mode="per-segment", inplace=False), "regular_ts"),
+            (StandardScalerTransform(in_column="target", mode="per-segment", inplace=True), "regular_ts"),
+            (YeoJohnsonTransform(in_column="target", mode="per-segment", inplace=False), "regular_ts"),
+            (YeoJohnsonTransform(in_column="target", mode="per-segment", inplace=True), "regular_ts"),
+        ],
+    )
+    def test_transform_train_subset_segments_failed_error(self, transform, dataset_name, request):
+        ts = request.getfixturevalue(dataset_name)
+        self._test_transform_train_subset_segments(ts, transform, segments=["segment_2"])
+
+
+class TestTransformFutureSubsetSegments:
+    """Test transform on future part of subset of segments.
+
+    Expected that transformation on subset of segments match subset of transformation on full dataset.
+    """
+
+    def _test_transform_future_subset_segments(self, ts, transform, segments, horizon=7):
+        # select subset of tsdataset
+        subset_ts = select_segments_subset(ts=deepcopy(ts), segments=segments)
+        train_df = ts.to_pandas()
+        ts.transforms = [transform]
+        subset_ts.transforms = [transform]
+
+        # fitting
+        transform.fit(train_df)
+
+        # transform full
+        transformed_future_ts = ts.make_future(future_steps=horizon)
+
+        # transform subset of segments
+        transformed_subset_future_ts = subset_ts.make_future(future_steps=horizon)
+
+        # checking
+        transformed_future_df = transformed_future_ts.to_pandas()
+        transformed_subset_future_df = transformed_subset_future_ts.to_pandas()
+        assert_frame_equal(transformed_subset_future_df, transformed_future_df.loc[:, pd.IndexSlice[segments, :]])
+
+    @pytest.mark.parametrize(
```

Comment:
I don't really think it is a good idea for now, because set of transforms, that works fine is different for each test. May be we can extract only some core, but it looks somewhat artifical.